src/server/scripts/Northrend/Ulduar/HallsOfStone/boss_maiden_of_grief.cpp

```diff
@@ -108,31 +109,35 @@ class boss_maiden_of_grief : public CreatureScript
                 case EVENT_STORM:
                     {
                         me->CastSpell(me->GetVictim(), DUNGEON_MODE(STORM_OF_GRIEF, STORM_OF_GRIEF_H), true);
-                        events.Repeat(10s);
+                        events.Repeat(16s, 20s);
                         break;
                     }
                 case EVENT_SHOCK:
                     {
                         me->CastSpell(me->GetVictim(), DUNGEON_MODE(SHOCK_OF_SORROW, SHOCK_OF_SORROW_H), false);
                         Talk(SAY_STUN);
 
-                        events.Repeat(16s, 22s);
+                        events.Repeat(19s, 33s);
                         break;
                     }
                 case EVENT_PILLAR:
                     {
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 50.0f, true, 0))
+                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 50.0f, true, true))
                             me->CastSpell(target, DUNGEON_MODE(PILLAR_OF_WOE, PILLAR_OF_WOE_H), false);
 
-                        events.Repeat(12s, 20s);
+                        events.Repeat(8s, 31s);
                         break;
                     }
                 case EVENT_PARTING:
                     {
-                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 50.0f, true, 0))
+                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 50.0f, [](Unit const* target)
+                        {
+                            return target && target->IsPlayer() && target->getPowerType() != POWER_MANA;
+                        }))
```

Comment:
```suggestion
                        if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, PowerUsersSelector(me, POWER_MANA, 50.0f, true)))
```

We have a selector for that (and yours currently only targets non-mana users)

Reply:
Oof derp moment fixed